### ISSUE TITLE
🤖 [AUTO] Implement Strategy: regression_tree_spy_daily_returns_extreme_leaves

### ISSUE THREAD

**Issue by kpakpooluwaseyi** [kpakpooluwaseyi/Agentic-Algo-Strategies#446]
## 🤖 Auto-Generated Strategy Request

**Source:** Artificial Intelligence Techniques 2017.pdf (strategy 3)
**Extracted:** 2025-12-25T02:45:26.358736

---

1.  **Strategy Name**: regression_tree_spy_daily_returns_extreme_leaves
2.  **Strategy Type**: Mean-reversion
3.  **Timeframe**: 1D (Holding for one day, entering at close, exiting at next close)
4.  **Instruments**: SPY (S&P 500 ETF)

5.  **Entry Rules**:
    *   **Long Entry**: Enter a long position if `ret2 < 1.53%` AND `ret1 < -1.39%`. (This corresponds to the leaf node with the highest expected positive return in the example.)
    *   **Short Entry**: Enter a short position if `ret2 >= 1.53%`. (This corresponds to the leaf node with the most negative expected return in the example.)

6.  **Exit Rules**:
    *   **Take Profit**: Not explicitly defined, but implied by 1-day holding period.
    *   **Stop Loss**: Not explicitly defined.
    *   **Time-based Exit**: All positions are held for one trading day.

7.  **Indicators Required**:
    *   Previous 1-day return (`ret1`)
    *   Previous 2-day return (`ret2`)
    *   Regression Tree (MATLAB `fitrtree` function, with `MinLeafSize` set to 100 in the example).
    *   The example tree used `ret1` and `ret2` primarily.

8.  **Risk Management**:
    *   **Position Sizing**: Fixed capital amount per trade.
    *   **Maximum Risk per Trade**: Not specified.
    *   **Risk-Reward Ratio**: Not specified.

9.  **Additional Notes**:
    *   This strategy is derived by analyzing the "extreme" leaf nodes of the regression tree for trading signals.
    *   The rules indicate a mean-reverting behavior.
    *   The training data was from Dec 22, 2004, to Sep 15, 2009. The test data was from Sep 16, 2009, to Jun 2, 2014.
    *   Performance: 28.8% CAGR (in-sample), 1.5 Sharpe (in-sample); 3.9% CAGR (out-of-sample), 0.5 Sharpe (out-of-sample).
    *   Using *all* leaves for prediction without overfitting reduction techniques leads to severe overfitting and poor out-of-sample results (-7.2% CAGR).

---

## Implementation Instructions

Please implement this strategy using the `backtesting.py` framework with the following requirements:

1. **File Location:** `strategies/regression_tree_spy_daily_returns_extreme_leaves.py`
2. **Data Path:** Use `data/BTC-USD-15m.csv` for backtesting
3. **Indicators:** Use `talib` or `pandas_ta` (NOT backtesting.py's built-in indicators)
4. **Template:** Follow existing strategies in the `strategies/` folder

### Required Components:
- Strategy class inheriting from `Strategy`
- Proper indicator initialization using `self.I()`
- Entry/exit logic in `next()` method
- Default parameters as class attributes
- Risk management (stop loss, take profit)

### Data Format:
```
datetime, open, high, low, close, volume
2023-01-01 00:00:00, 16531.83, 16532.69, 16509.11, 16510.82, 231.05
```

### Output:
- Run backtest with default parameters
- Print stats and create HTML plot
- Save results to `results/temp_result.json`
